apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts

```diff
@@ -66,4 +90,220 @@ describe('=> ManagerWithdrawHook', () => {
+      ).revertedWith(
+        `AccessControl: account ${user.address.toLowerCase()} is missing role ${await managerWithdrawHook.SET_DEPOSIT_RECORD_ROLE()}`
+      )
+    })
+
+    it('sets to non-zero address', async () => {
+      await managerWithdrawHook.connect(deployer).setDepositRecord(ZERO_ADDRESS)
+      expect(depositRecord.address).to.not.eq(ZERO_ADDRESS)
+      expect(await managerWithdrawHook.getDepositRecord()).to.not.eq(depositRecord.address)
+
+      await managerWithdrawHook.connect(deployer).setDepositRecord(depositRecord.address)
+
+      expect(await managerWithdrawHook.getDepositRecord()).to.eq(depositRecord.address)
+    })
+
+    it('sets to zero address', async () => {
+      expect(await managerWithdrawHook.getDepositRecord()).to.not.eq(ZERO_ADDRESS)
+
+      await managerWithdrawHook.connect(deployer).setDepositRecord(ZERO_ADDRESS)
+
+      expect(await managerWithdrawHook.getDepositRecord()).to.eq(ZERO_ADDRESS)
+    })
+
+    it('is idempotent', async () => {
+      await managerWithdrawHook.connect(deployer).setDepositRecord(ZERO_ADDRESS)
+      expect(await managerWithdrawHook.getDepositRecord()).to.not.eq(depositRecord.address)
+
+      await managerWithdrawHook.connect(deployer).setDepositRecord(depositRecord.address)
+
+      expect(await managerWithdrawHook.getDepositRecord()).to.eq(depositRecord.address)
+
+      await managerWithdrawHook.connect(deployer).setDepositRecord(depositRecord.address)
+
+      expect(await managerWithdrawHook.getDepositRecord()).to.eq(depositRecord.address)
+    })
+
+    it('emits DepositRecordChange', async () => {
+      const tx = await managerWithdrawHook.connect(deployer).setDepositRecord(depositRecord.address)
+
+      await expect(tx)
+        .to.emit(managerWithdrawHook, 'DepositRecordChange')
+        .withArgs(depositRecord.address)
+    })
+  })
+
+  describe('# setMinReservePercentage', () => {
+    beforeEach(async () => {
+      await getSignersAndDeployHook()
+      await grantAndAcceptRole(
+        managerWithdrawHook,
+        deployer,
+        deployer,
+        await managerWithdrawHook.SET_MIN_RESERVE_PERCENTAGE_ROLE()
+      )
+    })
+
+    it('reverts if not role holder', async () => {
+      expect(
+        await managerWithdrawHook.hasRole(
+          await managerWithdrawHook.SET_MIN_RESERVE_PERCENTAGE_ROLE(),
+          user.address
+        )
+      ).to.eq(false)
+
+      await expect(
+        managerWithdrawHook.connect(user).setMinReservePercentage(TEST_MIN_RESERVE_PERCENTAGE)
+      ).revertedWith(
+        `AccessControl: account ${user.address.toLowerCase()} is missing role ${await managerWithdrawHook.SET_MIN_RESERVE_PERCENTAGE_ROLE()}`
+      )
+    })
+
+    it('reverts if exceeds 100%', async () => {
+      await expect(
+        managerWithdrawHook.connect(deployer).setMinReservePercentage(RESERVE_DENOMINATOR + 1)
+      ).revertedWith('exceeds 100%')
+    })
+
+    it('sets to non-zero value < 100%', async () => {
```

Comment:
```suggestion
    it('sets to non-zero value', async () => {
```
the not >100% part is implied by above revert test